src/azul/deployment.py

```diff
@@ -380,6 +385,21 @@ def resource(self, *args, **kwargs):
         """
         return self.boto3_session.resource(*args, **kwargs)
 
+    def caller_user_id(self) -> Tuple[str, Optional[str]]:
+        """
+        Returns a tuple of the unique ID for the current AWS user. Some
+        principals, such as the AWS root user or an IAM user, dont have a second
+        value within the unique string, in these cases the second entry in the
+        tuple is `None`.
+        """
+        user = self.sts.get_caller_identity()['UserId']
+        try:
+            account, role_name = user.split(':')
+        except ValueError:
+            account = user
+            role_name = None
+        return account, role_name
```

Comment:
The first element is only an account (in the AWS sense) for federated users and root users and we don't use those case. The second element is never a role name. The docstring and the naming of this function and the variables seem to be totally inconsistent with each other and reality.

If you really want this function to be of general use, which you seem to be going for, you'd want to document the cases listed in the `aws:userid` column of the table in 

https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_variables.html#principaltable

I'd recommend against that, though. You'd be better off making the implementation, the naming and the documentation comprehensible and consistent for the sole case that applies to us:  assumed role.

Reply:
https://github.com/DataBiosphere/azul/issues/3890